packages/core/src/client.ts

```diff
@@ -606,6 +606,12 @@ export abstract class Client<O extends ClientOptions = ClientOptions> {
     ) => void,
   ): () => void;
 
+  /**
+   * A hook for the browser tracing integrations to trigger the end of a page load span.
+   * @returns {() => void} A function that, when executed, removes the registered callback.
+   */
+  public on(hook: 'endPageloadSpan', callback: () => void): () => void;
```

Comment:
I decided to go with the client hooks approach instead of exposing a method on `browserTracingIntegration` because in contrast to all other integrations, we don't hide the specific implementation of `browserTracingIntegration` behind `defineIntegration`. So adding a method would indeed be public API (we also can't introduce `defineIntegration` now as that would be a breaking change).

Reply:
IMHO this is a nice enough API for this anyhow, I like it!